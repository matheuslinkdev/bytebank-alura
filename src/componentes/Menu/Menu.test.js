import { render, screen } from '@testing-library/react';
import Menu from './index';

test('Must render a link to the home page', () => {
  render(<Menu />);
  const linkHomePage = screen.getByText('Inicial');
  expect(linkHomePage).toBeInTheDocument();
});

test('Must render a links list', () => {
  render(<Menu />);
  const linksList = screen.getAllByRole('link');
  expect(linksList).toHaveLength(4);
});

test('Must not render the statement link', () => {
  render(<Menu />);
  const statementLink = screen.queryByText('Extrato');
  expect(statementLink).not.toBeInTheDocument();
});

test('Must render a link list with class link', () => {
  render(<Menu />);
  const links = screen.getAllByRole('link');
  links.forEach((link) => expect(link).toHaveClass('link'));
  expect(links).toMatchSnapshot()
});
