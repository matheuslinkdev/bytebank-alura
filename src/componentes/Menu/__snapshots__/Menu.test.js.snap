// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must render a link list with class link 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
